bash_scripts/reencode_videos_sleap.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+#SBATCH -p gpu # partition
+#SBATCH -N 1   # number of nodes
+#SBATCH --mem 64G # memory pool for all cores
+#SBATCH -n 2 # number of cores
+#SBATCH -t 3-00:00 # time (D-HH:MM)
+#SBATCH --gres gpu:1 # request 1 GPU (of any kind)
+#SBATCH -o slurm_array.%N.%A-%a.out
+#SBATCH -e slurm_array.%N.%A-%a.err
+#SBATCH --mail-type=ALL
+#SBATCH --mail-user=s.minano@ucl.ac.uk
+#SBATCH --array=0-7%4
+
+# load modules
+module load SLEAP
+
+
+# input/output data dirs
+INPUT_DATA_LIST=(
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/09.08.2023-Day2/09.08.2023-01-Left.MOV"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/09.08.2023-Day2/09.08.2023-03-Left.MOV"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/09.08.2023-Day2/09.08.2023-04-Left.MOV"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/09.08.2023-Day2/09.08.2023-04-Right.MOV"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/10.08.2023-Day3/10.08.2023-01-Left.mov"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/10.08.2023-Day3/10.08.2023-01-Right.mov"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/10.08.2023-Day3/10.08.2023-02-Left.MOV"
+    "/ceph/zoo/raw/CrabField/ramalhete_2023/10.08.2023-Day3/10.08.2023-02-Right.MOV"
+
+)
+
+OUTPUT_DIR=/ceph/zoo/users/sminano/crabs_reencoding
+OUTPUT_SUBDIR="20230816_ramalhete2023_day2"
+
+echo "Input data:"
+for i in "${INPUT_DATA_LIST[@]}"
+do
+    echo "$i"
+done
+echo "Output dir: $OUTPUT_DIR/$OUTPUT_SUBDIR"
+echo "--------"
+
+# Check len(list of input data) matches max SLURM_ARRAY_TASK_COUNT
+# if not, exit
+if [[ $SLURM_ARRAY_TASK_COUNT -ne ${#INPUT_DATA_LIST[@]} ]]; then
+    echo "The number of array tasks does not match the number of inputs"
+    exit 1
+fi
+
+# reencode input videos following SLEAP's recommendations
+# https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format
+for i in {1..${SLURM_ARRAY_TASK_COUNT}}
+do
+    FILEPATH=${INPUT_DATA_LIST[${SLURM_ARRAY_TASK_ID}]}
+    filename_no_ext="$(basename "$FILEPATH" | sed 's/\(.*\)\..*/\1/')"
+    echo "Input video: $FILEPATH"
+
+    ffmpeg -y -i "$FILEPATH" \
+    -c:v libx264 \
+    -pix_fmt yuv420p \
+    -preset superfast \
+    -crf 15 \
+    "$OUTPUT_DIR/$OUTPUT_SUBDIR/$filename_no_ext.mp4"
```

Comment:
Do we want a [`-loglevel verbose`, or a `-v`](https://ffmpeg.org/ffmpeg.html#Generic-options)?

And do you `tee` the output (of this script) somewhere?
Actually -- and I'm showing my `slurm`-ignorance -- do you get that anyway in the job logs?

Reply:
> Do we want a -loglevel verbose, or a -v?

In the "upgrade" of this script, I reencode the video and then extract frames from there (PR #78). For that, I had a go at running `ffmpeg -v`, but it was a bit too verbose.... In fact the default info output level already prints out the parameters of the reencoding, so I ended up only adding a line with `ffmpeg --version` to make sure that was registered. I add now the `ffmpeg --version` line here, but I think this script and the reencode+extract frames one are likely better as one...

And yes, as you say I get`ffmpeg`'s stdout and stderr in slurm's logs (.err and .out files)